From: Jan-Ivar Bruaroey <jan-ivar@users.noreply.github.com>
Date: Fri, 21 Feb 2025 08:01:00 -0500
Subject: (tmp-cherry-pick) Revert "Remove rtc_p2p" (c6278957d1)

This reverts commit a05ad63aa336b22850387dafdcb5f7fcb00c7fb3.

Reason for revert: breaks importer :(

Original change's description:
> Remove rtc_p2p
>
> This completes the task of removing the rtc_p2p target.
>
> Bug: webrtc:42226155
> Change-Id: I2cd145fe250fc2e572b4cacc0982520dbf2250b8
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/375660
> Commit-Queue: Mirko Bonadei <mbonadei@webrtc.org>
> Auto-Submit: Jonas Oreland <jonaso@webrtc.org>
> Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#43817}

Bug: webrtc:42226155
Change-Id: If0ba5c78c376ed31a8cae48ea0fee7162f70d5ff
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/375720
Auto-Submit: Jonas Oreland <jonaso@webrtc.org>
Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
Commit-Queue: Mirko Bonadei <mbonadei@webrtc.org>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#43819}
---
 p2p/BUILD.gn | 161 ++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 160 insertions(+), 1 deletion(-)

diff --git a/p2p/BUILD.gn b/p2p/BUILD.gn
index 760b602c54..ec45e3684b 100644
--- a/p2p/BUILD.gn
+++ b/p2p/BUILD.gn
@@ -9,7 +9,10 @@
 import("../webrtc.gni")
 
 group("p2p") {
-  deps = [ ":libstunprober" ]
+  deps = [
+    ":libstunprober",
+    ":rtc_p2p",
+  ]
 }
 
 rtc_source_set("active_ice_controller_factory_interface") {
@@ -1226,3 +1229,159 @@ if (rtc_include_tests) {
     ]
   }
 }
+
+# The rtc_p2p target used to be a collection of a large number of headers
+# that were used by others, but those have been moved to their own targets,
+# so the headers listed here are legacy and should be removed.
+# The target is placed last so that automation will not choose it as a
+# source of headers.
+rtc_library("rtc_p2p") {
+  visibility = [ "*" ]
+  sources = [
+    # TODO(bugs.webrtc.org/15796): Finish cleanup and remove.
+    "base/active_ice_controller_factory_interface.h",  # To be removed (Chrome)
+    "base/active_ice_controller_interface.h",  # To be removed (Internal)
+    "base/basic_ice_controller.h",  # To be removed (Chrome)
+    "base/basic_packet_socket_factory.h",  # To be removed
+    "base/candidate_pair_interface.h",  # To be removed
+    "base/connection.h",  # To be removed
+    "base/connection_info.h",  # To be removed
+    "base/default_ice_transport_factory.h",  # To be removed (Chrome)
+    "base/dtls_transport.h",  # To be removed
+    "base/dtls_transport_internal.h",  # To be removed
+    "base/ice_agent_interface.h",  # To be removed (Chrome)
+    "base/ice_controller_interface.h",  # To be removed (Chrome)
+    "base/ice_credentials_iterator.h",  # To be removed
+    "base/ice_switch_reason.h",  # To be removed (Chrome)
+    "base/ice_transport_internal.h",  # To be removed
+    "base/p2p_constants.h",  # To be removed
+    "base/p2p_transport_channel.h",  # To be removed (Chrome)
+    "base/p2p_transport_channel_ice_field_trials.h",  # To be removed
+    "base/packet_transport_internal.h",  # To be removed
+    "base/port.h",  # To be removed (Chrome)
+    "base/port_allocator.h",  # To be removed (Chrome)
+    "base/port_interface.h",  # To be removed
+    "base/pseudo_tcp.h",  # To be removed
+    "base/stun_dictionary.h",  # To be removed
+    "base/stun_port.h",  # To be removed
+    "base/tcp_port.h",  # To be removed
+    "base/transport_description.h",  # To be removed
+    "base/transport_info.h",  # To be removed
+    "base/turn_port.h",  # To be removed
+    "client/basic_port_allocator.h",  # To be removed
+    "client/turn_port_factory.h",  # To be removed
+  ]
+
+  deps = [
+    ":active_ice_controller_factory_interface",
+    ":async_stun_tcp_socket",
+    ":basic_async_resolver_factory",
+    ":basic_ice_controller",
+    ":basic_packet_socket_factory",
+    ":candidate_pair_interface",
+    ":connection",
+    ":connection_info",
+    ":dtls_stun_piggyback_controller",
+    ":dtls_transport",
+    ":dtls_transport_internal",
+    ":ice_agent_interface",
+    ":ice_controller_factory_interface",
+    ":ice_controller_interface",
+    ":ice_credentials_iterator",
+    ":ice_switch_reason",
+    ":ice_transport_internal",
+    ":p2p_constants",
+    ":p2p_transport_channel",
+    ":p2p_transport_channel_ice_field_trials",
+    ":packet_transport_internal",
+    ":port",
+    ":port_allocator",
+    ":port_interface",
+    ":pseudo_tcp",
+    ":regathering_controller",
+    ":relay_port_factory_interface",
+    ":stun_dictionary",
+    ":stun_request",
+    ":tcp_port",
+    ":transport_description",
+    ":udp_port",
+    ":wrapping_active_ice_controller",
+    "../api:array_view",
+    "../api:async_dns_resolver",
+    "../api:candidate",
+    "../api:dtls_transport_interface",
+    "../api:field_trials_view",
+    "../api:ice_transport_interface",
+    "../api:make_ref_counted",
+    "../api:packet_socket_factory",
+    "../api:rtc_error",
+    "../api:scoped_refptr",
+    "../api:sequence_checker",
+    "../api:turn_customizer",
+    "../api/crypto:options",
+    "../api/rtc_event_log",
+    "../api/task_queue",
+    "../api/task_queue:pending_task_safety_flag",
+    "../api/transport:enums",
+    "../api/transport:field_trial_based_config",
+    "../api/transport:stun_types",
+    "../api/units:time_delta",
+    "../api/units:timestamp",
+    "../logging:ice_log",
+    "../rtc_base:async_dns_resolver",
+    "../rtc_base:async_packet_socket",
+    "../rtc_base:async_tcp_socket",
+    "../rtc_base:async_udp_socket",
+    "../rtc_base:buffer",
+    "../rtc_base:buffer_queue",
+    "../rtc_base:byte_buffer",
+    "../rtc_base:byte_order",
+    "../rtc_base:callback_list",
+    "../rtc_base:checks",
+    "../rtc_base:copy_on_write_buffer",
+    "../rtc_base:crc32",
+    "../rtc_base:dscp",
+    "../rtc_base:event_tracer",
+    "../rtc_base:ip_address",
+    "../rtc_base:logging",
+    "../rtc_base:macromagic",
+    "../rtc_base:mdns_responder_interface",
+    "../rtc_base:net_helper",
+    "../rtc_base:net_helpers",
+    "../rtc_base:network",
+    "../rtc_base:network_constants",
+    "../rtc_base:network_route",
+    "../rtc_base:rate_tracker",
+    "../rtc_base:refcount",
+    "../rtc_base:rtc_numerics",
+    "../rtc_base:safe_minmax",
+    "../rtc_base:socket",
+    "../rtc_base:socket_adapters",
+    "../rtc_base:socket_address",
+    "../rtc_base:socket_factory",
+    "../rtc_base:socket_server",
+    "../rtc_base:ssl",
+    "../rtc_base:ssl_adapter",
+    "../rtc_base:stream",
+    "../rtc_base:stringutils",
+    "../rtc_base:threading",
+    "../rtc_base:timeutils",
+    "../rtc_base:weak_ptr",
+    "../rtc_base/containers:flat_map",
+    "../rtc_base/experiments:field_trial_parser",
+    "../rtc_base/memory:always_valid_pointer",
+    "../rtc_base/network:received_packet",
+    "../rtc_base/network:sent_packet",
+    "../rtc_base/synchronization:mutex",
+    "../rtc_base/system:no_unique_address",
+    "../rtc_base/system:rtc_export",
+    "../rtc_base/third_party/base64",
+    "../rtc_base/third_party/sigslot",
+    "../system_wrappers:metrics",
+    "//third_party/abseil-cpp/absl/algorithm:container",
+    "//third_party/abseil-cpp/absl/base:core_headers",
+    "//third_party/abseil-cpp/absl/functional:any_invocable",
+    "//third_party/abseil-cpp/absl/memory",
+    "//third_party/abseil-cpp/absl/strings:string_view",
+  ]
+}
